src/components/Chat.js

```diff
@@ -88,20 +90,40 @@ function Chat({
   );
   const messages = useFirebaseQuery(messagesQuery);
 
-  function handleSubmit(event) {
+  async function handleSubmit(event) {
     event.preventDefault();
     if (input) {
       if (filter.isProfane(input)) {
         alert(
           "We detected that your message contains profane language. If you think this was a mistake, please let us know!"
         );
       } else {
-        firebase.database().ref(databasePath).push({
-          user: user.id,
-          message: input,
-          time: firebase.database.ServerValue.TIMESTAMP,
-        });
-        setInput("");
+        let ref = await firebase
+          .database()
+          .ref(`users/${user.id}/canChat`)
+          .once("value");
+        let canChat = ref.val();
+        if (!canChat) {
+          ref = await firebase
+            .database()
+            .ref(`/userGames/${user.id}`)
+            .once("value");
```

Comment:
Can we add a `limit(MIN_GAMES)` here, to reduce the burden on the database if possible?